packages/core-data/src/resolvers.js

```diff
@@ -140,43 +140,61 @@ export function* canUser( action, resource, id ) {
 		delete: 'DELETE',
 	};
 
-	const method = methods[ action ];
-	if ( ! method ) {
-		throw new Error( `'${ action }' is not a valid action.` );
+	const isCustomAction = ! methods.hasOwnProperty( action );
+
+	let path = '/wp/v2/' + resource;
+
+	if ( id ) {
+		path += '/' + id;
```

Comment:
`resource` should be considered a required parameter. `id` is optional†, but this concatenation occurs in a condition to consider it as being set.

_† The JSDoc should be updated from the nullable `{?string}` type to the [correct `[id]` optional parameter notation](https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#nullable-undefined-and-void-types)._

Reply:
What if I want to request the root/index resource?

Reply:
You would omit the `id` parameter.

Reply:
I think I might not be understanding what you're asking. There will always be a resource.

When I say it doesn't necessarily need to be a _singular_ resource, I consider it the same as how it might map to a use of `current_user_can` in PHP:

```php
current_user_can( 'publish_posts' );
current_user_can( 'publish_post', $post->ID );
```

https://developer.wordpress.org/reference/functions/current_user_can/

I see this mapping to `canUser` as something like:

```js
canUser( 'publish', 'posts' );
canUser( 'publish', 'posts', postId );
```

The caveat mentioned in the original post is that, currently, the REST API only supports the latter of these two, since it only includes action `_links` on the response of an item:

https://github.com/WordPress/wordpress-develop/blob/47e38ea/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L1675-L1683

My thinking is that, to represent the first of these (without an ID), it _could_ be possible to include the action `_links` properties on the collection response as well:

https://github.com/WordPress/wordpress-develop/blob/47e38ea/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L155

Reply:
Right, but we also have a base resource:

https://github.com/WordPress/gutenberg/commit/f6ce79cd681754a1f19714e8c8e1f65d1cd19b57#diff-c5eaec560b039f2dc76f93fa76ba89da

Aren't there capabilities for `/wp/v2/` or `/`?

I still think that `canUser` should have the same API as `getEntityRecord` with `kind` and `name` and we can just grab the resource from the entity config as I outlined here: https://github.com/WordPress/gutenberg/pull/18956#pullrequestreview-327924547.

Reply:
> Aren't there capabilities for `/wp/v2/` or `/`?

Doesn't seem to me like there would be.

> I still think that `canUser` should have the same API as `getEntityRecord` with `kind` and `name` and we can just grab the resource from the entity config as I outlined here: [#18956 (review)](https://github.com/WordPress/gutenberg/pull/18956#pullrequestreview-327924547).

I agree this should be a goal for reuse. Not entirely sure yet how I feel about the overloaded signature, or if it's something to reevaluate whether "resource" is _ever_ a sufficient parameter (i.e. might be more inclined to deprecate this parameter in favor of `kind`, `name`, if it came to it, or perhaps vice versa).

Reply:
Oh, I never meant to have an overloaded signature. I was suggesting to deprecate "resource".